TEveEventDisplay/src/TEveMu2eHit.cc

```diff
@@ -50,58 +106,35 @@ namespace mu2e{
   }
 
   /*------------Function to 2D draw hits:-------------*/
-  void TEveMu2eHit::DrawHit2D(const std::string &pstr, Int_t n, CLHEP::Hep3Vector pointInMu2e, int energylevel, TEveElementList *HitList2DXY, TEveElementList *HitList2DXZ)
-  { 
-    mu2e::GeomHandle<mu2e::Tracker> tracker;
-    const auto& allStraws = tracker->getStraws();
-    for (size_t i = 0; i<tracker->nStraws(); i++)
-      {
-      const mu2e::Straw& s = allStraws[i];
-      const CLHEP::Hep3Vector& p = s.getMidPoint();
-      const CLHEP::Hep3Vector& d = s.getDirection();
-      double x = p.x();
-      double y = p.y();
-      double z = p.z();
-      double theta = d.theta();
-      double phi = d.phi();
-      double l = s.halfLength();
-      double st=sin(theta);
-      double ct=cos(theta);
-      double sp=sin(phi+TMath::Pi()/2.0);   
-      double cp=cos(phi+TMath::Pi()/2.0);
-      double x1=x+l*st*sp;
-      double y1=y-l*st*cp;
-      double z1=z+l*ct;  
-      double x2=x-l*st*sp;
-      double y2=y+l*st*cp;
-      double z2=z-l*ct;
-      int a = x1*(y2-pointInMu2e.y()) + x2*(pointInMu2e.y()-y1) + pointInMu2e.x()*(y1-y2);
-      if(a == 0)
-        {
-        TEveMu2eCustomHelix *line_twoDstrawXY = new TEveMu2eCustomHelix();
-        line_twoDstrawXY->SetLineWidth(1);
-        line_twoDstrawXY->SetPoint(0,pointmmTocm(x1),pointmmTocm(y1),pointmmTocm(z1));
-        line_twoDstrawXY->SetNextPoint(pointmmTocm(x2),pointmmTocm(y2),pointmmTocm(z2));
-        line_twoDstrawXY->SetLineColor(kRed);
-        HitList2DXY->AddElement(line_twoDstrawXY);
-        
-        TEveMu2eCustomHelix *line_twoDstrawXZ = new TEveMu2eCustomHelix();
-        line_twoDstrawXZ->SetLineWidth(1);
-        line_twoDstrawXZ->SetPoint(0,pointmmTocm(x1),pointmmTocm(y1)+ 1000,pointmmTocm(z1));
-        line_twoDstrawXZ->SetNextPoint(pointmmTocm(x2),pointmmTocm(y2)+ 1000,pointmmTocm(z2));
-        line_twoDstrawXZ->SetLineColor(kRed);
-        HitList2DXZ->AddElement(line_twoDstrawXZ);
-        }
-      }
+ void TEveMu2eHit::DrawHit2D(const std::string &pstr, Int_t n, CLHEP::Hep3Vector pointInMu2e, int energylevel, TEveElementList *HitList2DXY, TEveElementList *HitList2DXZ)
+  {
+    auto [sposi, sposf, title, colorid] = DrawStraw();
+    if(sposi.x()!=0){
+      TEveMu2eCustomHelix *line_twoDstrawXY = new TEveMu2eCustomHelix();
+      line_twoDstrawXY->SetLineWidth(1);
+      line_twoDstrawXY->SetPoint(0,pointmmTocm(sposi.x()),pointmmTocm(sposi.y()),pointmmTocm(sposi.z()));
+      line_twoDstrawXY->SetNextPoint(pointmmTocm(sposf.x()),pointmmTocm(sposf.y()),pointmmTocm(sposf.z()));
```

Comment:
In  TEveMu2eCustomHelix define overloads of these functions that take a vector type so that you do not need to write everything out in components every time.  It's harder to be sure that the code is correct.

Reply:
Will add to the list